### DISCUSSION TITLE
javascript problem

### DISCUSSION THREAD

**Discussion by mohammedawez** [Jay-study-nildana/FrontEndForStudents#37] [Coding Questions]
hlo jay sir....!
@Jay-study-nildana 

 i m facing with some issue in javascript code can u please solve , unable to see output in console .
error displayed as :-Uncaught ReferenceError: set is not defined

here is the file repositve file code link  :-
https://github.com/mohammedawez/problem-of-javascript

url:-
https://github.com/mohammedawez/problem-of-javascript.git

@Jay-study-nildana

**Comment by Jay-study-nildana** [comment]
@mohammedawez 

ah, alright. I will look into it today. just got up and starting my day. 

(Note: You should also put this code in CodePen along with GitHub)

Or 

(Note: since it is vanilla JS project, you don’t need to use GitHub at all and use only CodePen for such things)

I will update this thread once I have more information. 

Thanks.

Jay.

**Comment by Jay-study-nildana** [comment]
@mohammedawez 

So, first step, I have put it in  CodePen. 

https://codepen.io/jay-pancodu/pen/RwErVjm?editors=0011

And the error is happening in this specific line. 

```

new set(ratingData.map(({restaurant}) => restaurant))); 

```

Now, you are using a very advance feature called ‘set’ for some reason. Even I don’t use this advance feature so I am very confused why you are using it. 

more details about set (an advanced object concept) can be found here. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/set

So, I think, you found this code online somewhere and tried to copy paste. Tell me where you got this code online and I will try to read it. 

Another option is, this is a very simple puzzle. You should be using regular basic simple javascript functions to solve it. 

So, there you go. I hope this helps. 

Thanks. 

Jay

**Reply by mohammedawez** [reply]
no sir i have not found this code in online somewhere but in edureka they thought us ,i think you know bettter about it ,i should subbmit as assingment , i have sawed recoreding and coded it ,i dont know how should to be  solve it  ,i am not getting any output???????? how can get solved it??????


@Jay-study-nildana 
@Jay-study-nildana

**Reply by Jay-study-nildana** [reply]
@mohammedawez 

note: There is nothing wrong with taking code from online. It’s actually good if you got code from online :) 

. I don’t know awez. I have not used this set concept in my knowledge. Sorry brotha. I hope you can find the answer elsewhere. 

Thanks

Jay

**Reply by jonboribbins** [reply]
If you utilize code from online and tweak it a little you can learn a lot from it. There are multiple places where you could ask to find your answer. You may find help on a discord server, twitter, or meet up group as well.

Are you calculating the average rating for different restaurants and then filter out the restaurants with an average rating greater than or equal to 4?

const uniqueRestaurants = new Set(ratingData.map(({ restaurant }) => restaurant));

The symbols that are corrected look different now in this version. Hopefully, this helped.

**Reply by Jay-study-nildana** [reply]
@jonboribbins 

I dunno who you are brotha, but, thank you for jumping in.

Jay

**Reply by mohammedawez** [reply]
@jonboribbins 

thank you soo much  for this Answer buddy.......!!!!

@Jay-study-nildana  
and thanks for jay sir also......!!!

**Reply by Jay-study-nildana** [reply]
@jonboribbins 

I just realized, the error was an extra bracket! 

This isn't the first time I have been stumped with a bracket mismatch. From a young age, like 10 or something, I always open and close brackets immediately. As a consequence, i frequently am unable to diagnose bracket mismatch as I never encountered it in my own life. 

Still, I should have noticed it. Once again, thank you. 

@mohammedawez 

I am begging you, please open and close brackets, quotes, and tags immediately my friend. It is my humble request. 

Jay.

**Reply by mohammedawez** [reply]
@Jay-study-nildana 

sure sir......!!!!!!.

okay

**Reply by Jay-study-nildana** [reply]
@mohammedawez 

Excellente. 

Jay.
